docs/spec_draft.md

```diff
@@ -327,6 +328,65 @@ logical operation.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.broadcast_in_dim
+
+### Semantics
+
+Expands the dimensions and/or rank of an input tensor by duplicating the data
+in the `operand` tensor and produces a `result` tensor. Formally,
+`result[i0, i1, ..., iR-1]` $=$ `operand[j0, j1, ..., jR-1]` such that
+`jk` $=$ `dim(operand, k) == 1 ? 0 : i[broadcast_dimensions[k]]`, for all
+dimensions `k` in `operand`.
+
+### Operands
+
+| Name                   | Type                                  |
+|------------------------|---------------------------------------|
+| `operand`              | tensor of any supported element types |
+| `broadcast_dimensions` | 1-dimensional array of type `si64`    |
+
+### Results
+
+| Name     | Type                                  |
+|----------|---------------------------------------|
+| `result` | tensor of any supported element types |
+
+### Constraints
+
+  * (C1) `operand` and `result` have the element type.
+  * (C2) size(`broadcast_dimensions`) $=$ rank(`operand`).
+  * (C3) $0 \le$ `broadcast_dimensions`[i] $\lt$ rank(`result`) for all
+  dimensions i in `operand`.
+  * (C4) All dimensions in `broadcast_dimensions` are unique.
+  * (C5) For all dimensions `i` in `operand`:
```

Comment:
Let's unify `i`, `j`, and `k` to match the variables from the semantics. E.g. `result[i]` is used in semantics section but the same `i` is used in `dim(operand, i)`.